src/app/pages/Friends/Friends.tsx

```diff
@@ -0,0 +1,207 @@
+                  }
+                  link="#"
+                  avatar={`../../src/assets/avatars/${index}.webp`}
+                  buttonText="Unfollow"
+                  buttonStyle="warning"
+                  onClick={() => handleFriendClick(user.id)}
+                  key={user.id}
+                />
+              ))}
+
+            {friendsData.length === 0 &&
+              isLoading === false &&
+              'There is no Knight in your list :( Wanna add your first Knight?'}
+            {isLoading === true && 'Loading your Knights...'}
+          </section>
+          {showToTop === true && (
+            <div className={styles.news__toDo}>
+              <ToTop color="var(--clr-white)" />
+              <span>
+                This is everything :)
+                <br />
+                Come back later for new great stuff!
+              </span>
+            </div>
+          )}
+        </main>
+        <Navigation active="friends" />
+      </section>
+      {modalClosed === false && (
+        <div id="modal" className={styles.modal}>
+          <Header
+            textThin="Add"
+            textBold="Knight"
+            icon="close"
+            onClick={() => modalClick()}
+          />
+          <section className={styles.modal__section}>
+            <form className={styles.form} onSubmit={handleSubmit}>
+              <input
+                className={styles.form__input}
+                type="text"
+                placeholder="Type in your Knights Name"
+                value={inputValue}
+                onChange={(event) => setInputValue(event.target.value)}
+              />
+            </form>
+            {user === 'error' && (
+              <span className={styles.modal__span}>
+                There is no Knight with this Name... :(
+                <br />
+                Maybe try another one :)
+              </span>
+            )}
+
+            {result === true &&
+              user !== 'error' &&
+              user !== '' &&
+              Object.values(friendsData).includes(user[0].id) === false && (
+                <ProfileItem
+                  username={user[0].data.name}
+                  wins={
+                    user[0].data.global_stats !== null
+                      ? user[0].data.global_stats?.solo.placetop1
+                      : '0'
+                  }
+                  games={
+                    user[0].data.global_stats !== null
+                      ? user[0].data.global_stats.solo.matchesplayed
+                      : '0'
+                  }
+                  buttonStyle="success"
+                  buttonText="Follow"
+                  onClick={() => {
+                    handleFriendClick(user[0].id);
+                    setModal(true);
+                    setInputValue('');
+                    setUser('');
+                  }}
+                  link="#"
+                  avatar={'../../src/assets/avatars/5.webp'}
+                />
+              )}
```

Comment:
This looks like duplicate code

Reply:
The buttonStyle and buttonText props are different dependent if the queried user is already a friend or not.